badges/dashboard/templatetags/projects_tags.py

```diff
@@ -1,9 +1,24 @@
 from django import template
 from django.template.defaultfilters import mark_safe
+from subprocess import Popen, PIPE
 
 register = template.Library()
 
 
 @register.filter
 def purge_content(value):
-    return mark_safe(value.replace('<p>&nbsp;</p>', ''))
\ No newline at end of file
+    return mark_safe(value.replace('<p>&nbsp;</p>', ''))
+
+@register.filter
+def html2text(value):
+    """
+    Pipes given HTML string into the text browser W3M, which renders it.
+    Rendered text is grabbed from STDOUT and returned.
+    """
+    try:
+        cmd = "w3m -dump -T text/html -O ascii"
+        proc = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE)
+        return proc.communicate(str(value))[0]
+    except OSError:
+        # something bad happened, so just return the input
+        return value
```

Comment:
Have a look how text content is generated from HTML emails for the Mechanical MOOC: https://github.com/p2pu/mechanical-mooc/blob/master/mail/views.py#L22-L27.